src/main/java/org/cbioportal/domain/sample/Sample.java

```diff
@@ -1,18 +1,93 @@
 package org.cbioportal.domain.sample;
 
-import org.cbioportal.legacy.model.Patient;
+import org.cbioportal.domain.patient.Patient;
 
 import java.io.Serializable;
 import java.util.Objects;
 
-public record Sample (Integer internalId, String stableId, SampleType sampleType, Integer patientId,
-                      String patientStableId, Patient patient,
-                      String cancerStudyIdentifier, Boolean sequenced, Boolean copyNumberSegmentPresent,
-                      String uniqueSampleKey, String uniquePatientKey) implements Serializable {
+public record Sample (
+    Integer internalId,
+    String stableId,
+    SampleType sampleType,
+    Integer patientId, 
+    String patientStableId,
+    Patient patient,
+    String cancerStudyIdentifier,
+    Boolean sequenced,
+    Boolean copyNumberSegmentPresent, 
+    String uniqueSampleKey,
+    String uniquePatientKey
+) implements Serializable {
 
-    public Sample(Integer internalId, String stableId, String patientStableId, String cancerStudyIdentifier){
-        this(internalId,stableId,null,null,patientStableId,null, cancerStudyIdentifier, null, null,null, null);
+    public Sample(Integer internalId, String stableId, String patientStableId, String cancerStudyIdentifier) {
+        this(
+            internalId,
+            stableId,
+            (SampleType) null,
+            null,
+            patientStableId,
+            null,
+            cancerStudyIdentifier,
+            null,
+            null,
+            null,
+            null
+        );
     }
+
+    public Sample(
+        Integer internalId,
+        String stableId,
+        String sampleType,
+        Integer patientId,
+        String patientStableId,
+        String cancerStudyIdentifier,
+        String uniqueSampleKey,
+        String uniquePatientKey
+    ) {
+        this(
+            internalId,
+            stableId,
+            SampleType.fromString(sampleType),
+            patientId,
+            patientStableId,
+            null,
+            cancerStudyIdentifier,
+            null,
+            null,
+            uniqueSampleKey,
+            uniquePatientKey
+        );
+    }
+
+    public Sample(
```

Comment:
We should have a comment here explaining the difference between these Sample overloads